target.c

```diff
@@ -336,11 +336,14 @@ int tcmu_add_dev_to_recovery_list(struct tcmu_device *dev)
 		goto done;
 	}
 	list_add(&tpg_recovery_list, &tpg->recovery_entry);
+	rdev->flags |= TCMUR_DEV_FLAG_IN_RECOVERY;
```

Comment:
I think this should be done after adding `rdev` to `tpg->devs` list:
```
add_to_list: 
        list_add(&tpg->devs, &rdev->recovery_entry);
        rdev->flags |= TCMUR_DEV_FLAG_IN_RECOVERY;
```
Otherwise `TCMUR_DEV_FLAG_IN_RECOVERY` wouldn't get set in the "another device already kicked off recovery" case (or, if it somehow would, it's really not obvious).